kite-data/kite-data-core/src/test/java/org/kitesdk/data/spi/filesystem/TestFileSystemWriters.java

```diff
@@ -95,6 +96,43 @@ public void testBasicWrite() throws IOException {
     Assert.assertEquals("Should match written records",
         written, Lists.newArrayList((Iterator) init(reader)));
   }
+  
+  @Test
+  public void testWriteWithOldSchema() throws IOException {
+    Schema writerSchema = SchemaBuilder.record("Message").fields()
```

Comment:
This schema isn't compatible with `TEST_SCHEMA`, which is the dataset's schema. The required string field "message" is missing. This works because you're passing this `writerSchema` into `newReader`.

The problem is that I'm not sure this is testing something well-defined. It validates that this `writerSchema` as passed to `FileSystemWriter.newWriter` is used for the file (otherwise writes would fail) but I don't see the value of using it as the read schema because we expect that it is a valid read schema for both `TEST_SCHEMA` and itself.

I think this should pass the `writerSchema` you have here to `newWriter`, create a file, and validate that the file's schema is actually this one and not `TEST_SCHEMA`. At the writer level, that's what we are interested in. The other behavior (visible/hidden files and other checks) is validated in other tests and can probably be removed.

I also think there needs to be an overall test from the view layer. The motivation is to show that a view with an older write schema works. So I would like to see a view test that creates a dataset with a 2-field schema (one optional field) and writes through a view with a write-compatible schema that is missing the optional field. That validates then end-to-end behavior that we want.